src/main/java/uk/gov/hmcts/reform/sscs/service/DwpAddressLookupService.java

```diff
@@ -85,21 +86,29 @@ private Address buildAddress(OfficeAddress dwpAddress) {
     }
 
     public String getDwpRegionalCenterByBenefitTypeAndOffice(String benefitType, String dwpIssuingOffice) {
-
         Optional<OfficeMapping> officeMapping = getDwpMappingByOffice(benefitType, dwpIssuingOffice);
-
-        return PIP.name().equalsIgnoreCase(benefitType)
-                ? officeMapping.map(mapping -> mapping.getMapping().getDwpRegionCentre()).orElse(null) :
-                officeMapping.map(mapping -> mapping.getMapping().getCcd()).orElse(null);
+        return getDwpRegionalCenterByBenefitType(benefitType, officeMapping);
     }
 
     public String getDefaultDwpRegionalCenterByBenefitTypeAndOffice(String benefitType) {
-
         Optional<OfficeMapping> officeMapping = getDefaultDwpMappingByBenefitType(benefitType);
+        return getDwpRegionalCenterByBenefitType(benefitType, officeMapping);
+    }
 
-        return PIP.name().equalsIgnoreCase(benefitType)
-                ? officeMapping.map(mapping -> mapping.getMapping().getDwpRegionCentre()).orElse(null) :
-                officeMapping.map(mapping -> mapping.getMapping().getCcd()).orElse(null);
+    private String getDwpRegionalCenterByBenefitType(String benefitType, Optional<OfficeMapping> officeMapping) {
+        Benefit benefit;
+        try {
+            benefit = Benefit.getBenefitByCode(benefitType);
+        } catch (BenefitMappingException e) {
+            return null;
+        }
+        switch (benefit) {
```

Comment:
Instead of the switch statement, we could put the responsibility into Benefit similar to what has been done for the airlookup in https://github.com/hmcts/sscs-common/pull/808/files#diff-353344bfbdfb4e18374d0376dd26e71a5d87d761ad77cf1b930b9ec74a7413acR231